src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -197,8 +197,10 @@ private static SyntaxNode WithBlockBody(SyntaxNode node, BlockSyntax body)
         {
             declarationStatement = declarationStatement.WithAdditionalAnnotations(Formatter.Annotation);
 
-            var oldOutermostBlock = block;
-            var matches = FindMatches(document, expression, document, oldOutermostBlock, allOccurrences, cancellationToken);
+            // If we're within a local function, our scope for the new local declaration is expanded to include the enclosing member.
+            var scope = block.GetAncestor<LocalFunctionStatementSyntax>() == null ? block : (SyntaxNode)block.GetAncestor<MemberDeclarationSyntax>();
+
+            var matches = FindMatches(document, expression, document, scope, allOccurrences, cancellationToken);
```

Comment:
What determines where the eventual local gets generated? Will it always be in the member declaration now? Should we be finding the nearest common ancestor member/local function of all discovered matches and putting it there?

Reply:
(Maybe it all works great and I'm just asking for more tests on these cases.)

Reply:
Added more tests for these cases. Thank you!